MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs

```diff
@@ -622,6 +671,11 @@ private static DataFeedValueField<T> GenerateValueField<T>(IReadOnlyList<string>
             InitBase(valueField, path, configKey);
             valueField.InitSetupValue(field => field.SyncWithConfigKey(configKey, ConfigKeyChangeLabel));
 
+            if (GenericTypesAttribute.GetTypes(GenericTypesAttribute.Group.EnginePrimitives).Contains(typeof(T)) || typeof(T) == typeof(Type))
```

Comment:
You can replace the first part in the if with `Coder.IsEnginePrimitive(typeof(T))`